### ISSUE TITLE
Custom message fails for string_boundary_linter(), gives 'NA'

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2636]
```r
lint('grepl("\\\\^", spec)\n', string_boundary_linter())
# <text>:1:7: warning: [string_boundary_linter] NA Doing so is more readable and more efficient.
# grepl("\\^", spec)
#       ^~~~~
```

**Comment by MichaelChirico** [comment]
Actually looking more closely, this appears to be a false positive of `string_boundary_linter()`... `grepl("\\^", x)` is not the same as `startsWith(x, "^")`. h/t @olivroy

**Comment by olivroy** [comment]
Indeed  `grepl("\\^", x)` should be linted as fixed regex linter ;)
